### PR TITLE
[release-4.4] Bug 1795163: waits for extension-apiserver-authentication before rolling out a new version

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

File: pkg/operator/workloadcontroller/workload_controller.go [hunk 1]
```diff
@@ -157,6 +162,28 @@ func (c OpenShiftAPIServerOperator) sync() error {
...
+		if err != nil {
+			return err
+		}
+
+		if len(authConfigMap.Data["requestheader-client-ca-file"]) == 0 {
+			klog.V(2).Infof("waiting for requestheader-client-ca-file filed in %q configmap to be populated", "extension-apiserver-authentication")
+			// will be requeued by kubeInformersForKubeSystemNamespace informer
+			return nil
+		}
+		c.haveObservedExtensionConfigMap = true
```
Comment (sttts) [inline pkg/operator/workloadcontroller/workload_controller.go]: this races with more than 1 worker.

File: pkg/operator/workloadcontroller/workload_controller.go [hunk 2]
```diff
@@ -157,11 +162,33 @@ func (c OpenShiftAPIServerOperator) sync() error {
...
+	}
+
 	return syncOpenShiftAPIServer_v311_00_to_latest(c, operatorConfig)
 }
 
 // Run starts the openshift-apiserver and blocks until stopCh is closed.
-func (c *OpenShiftAPIServerOperator) Run(ctx context.Context, workers int) {
+func (c *OpenShiftAPIServerOperator) Run(ctx context.Context) {
```
Comment (sttts) [inline pkg/operator/workloadcontroller/workload_controller.go]: haha, workers were ignored.

Comment (p0lyn0mial) [reply pkg/operator/workloadcontroller/workload_controller.go]: fun, yeah, I thought you knew that :)

**Pull request merged**